rapids-cmake/cpm/patches/libcudacxx/install_rules.diff

```diff
@@ -0,0 +1,66 @@
+diff --git a/cmake/libcudacxxInstallRules.cmake b/cmake/libcudacxxInstallRules.cmake
+index 446ccb50..ff622bb7 100644
+--- a/cmake/libcudacxxInstallRules.cmake
++++ b/cmake/libcudacxxInstallRules.cmake
+@@ -1,5 +1,5 @@
+ option(libcudacxx_ENABLE_INSTALL_RULES
+-  "Enable installation of libcudacxx" ${libcudacxx_TOPLEVEL_PROJECT}
++  "Enable installation of libcudacxx" ${LIBCUDACXX_TOPLEVEL_PROJECT}
+ )
+ 
+ if (NOT libcudacxx_ENABLE_INSTALL_RULES)
+@@ -12,24 +12,27 @@ include(GNUInstallDirs)
+ # Libcudacxx headers
+ install(DIRECTORY "${libcudacxx_SOURCE_DIR}/include/cuda"
+   DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
++  PATTERN CMakeLists.txt EXCLUDE
+ )
+ install(DIRECTORY "${libcudacxx_SOURCE_DIR}/include/nv"
+   DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}"
++  PATTERN CMakeLists.txt EXCLUDE
+ )
+ 
+ # Libcudacxx cmake package
+ install(DIRECTORY "${libcudacxx_SOURCE_DIR}/lib/cmake/libcudacxx"
+   DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake"
+-  PATTERN libcudacxx-header-search EXCLUDE
++  PATTERN *.cmake.in EXCLUDE
+ )
+ 
+ # Need to configure a file to store CMAKE_INSTALL_INCLUDEDIR
+ # since it can be defined by the user. This is common to work around collisions
+ # with the CTK installed headers.
++set(install_location "${CMAKE_INSTALL_LIBDIR}/cmake/libcudacxx")
```

Comment:
Why was it important to have this variable set before the below install commands rather than hardcoding it in the install?

Reply:
The variable is also expanded by the `configure_file` of `libcudacxx-header-search.cmake"`.